tf2_ros/src/transform_listener.cpp

```diff
@@ -54,19 +54,24 @@ TransformListener::TransformListener(tf2::BufferCore & buffer, bool spin_thread)
 
 TransformListener::~TransformListener()
 {
-  stop_thread_ = true;
+  if (spin_thread_) {
```

Comment:
Is this boolean actually necessary? Could we just use the `unique_ptr`'s value and eliminate the extra value?
```suggestion
  if (executor_) {
```

Reply:
Agreed. Pushed a commit to remove the extra boolean.